src/test/java/org/terracotta/statistics/derived/latency/DefaultLatencyHistogramStatisticTest.java

```diff
@@ -0,0 +1,195 @@
+    int[] BUCKETS = {10, 20, 50};
+    Duration[] WINDOWS = {ofMinutes(1), ofMinutes(5), ofMinutes(10), ofMinutes(30), ofMinutes(60), ofMinutes(120)};
+
+    // Generates Junit's parameters
+    return DoubleStream.of(PHIS).boxed().flatMap(phi ->
+        IntStream.of(BUCKETS).boxed().flatMap(buckets ->
+            Stream.of(WINDOWS).map(window ->
+                new Object[]{window, buckets, phi}
+            )))
+        .collect(Collectors.toList());
+  }
+
+  @BeforeClass
+  public static void simulateOperations() {
+    long seed = System.nanoTime();
+    System.out.println("seed: " + seed); // keep this in case test fails so that we can reproduce the failure
+
+    //OPERATIONS = LatencyUtils.generateLatencies(new Random(seed), 15, OPERATIONS_DURATION);
+    OPERATIONS = LatencyUtils.generateLatencies(new Random(seed), OPS_PER_SEC, OPERATIONS_DURATION, MEAN_LATENCY.toNanos());
+    System.out.println("ops: " + OPERATIONS.size());
+    assertThat((double) OPERATIONS.size(), is(closeTo(OPERATIONS_DURATION.getSeconds() * OPS_PER_SEC, OPERATIONS_DURATION.getSeconds() * OPS_PER_SEC * .20)));
+    OPERATIONS.forEach(sample -> {
+      assertThat(sample.time(), greaterThanOrEqualTo(0L));
+      assertThat(sample.value(), greaterThan(0L));
+    });
+
+    PERCENTILES = getPercentiles(OPERATIONS, identity());
+  }
+
+  private final DefaultLatencyHistogramStatistic statistic;
+  private final long window;
+  private final double phi;
+  private long now = Time.time();
+  private List<LongSample> lastOperations;
+  private Map<String, Percentile> lastPercentiles;
+
+  public DefaultLatencyHistogramStatisticTest(Duration window, int bucketCount, double phi) {
+    statistic = new DefaultLatencyHistogramStatistic(phi, bucketCount, window, () -> now);
+    this.window = window.toNanos();
+    this.phi = phi;
+  }
+
+  @Before
+  public void init() {
+    // will do some assertions during the insertion each 50% of the window
+    fillDerivedStatistic(window / 2, insertedSamples -> {
+
+      LongSample lastInsertedSample = insertedSamples.get(insertedSamples.size() - 1);
+
+      Map<String, Percentile> percentiles = getPercentiles(statistic, identity());
+      long max = round(percentiles.get("max").value());
+
+      long end = lastInsertedSample.time(); // inclusive
+      long start = end - window; // exclusive
+
+      String reason = "Start: " + start +
+          "\nEnd: " + end +
+          "\nMax: " + max +
+          "\nHistogram: " + statistic +
+          "\nContent: " + statistic.buckets();
+
+      long count = insertedSamples.stream()
+          .filter(sample -> sample.time() > start && sample.time() <= end)
+          .peek(sample -> {
+            if (phi < 1.5) {
+              // with a high Phi, high percentile values are not accurate at all
+              assertThat(reason + "\nSample: " + sample, sample.value(), is(lessThanOrEqualTo(max)));
```

Comment:
why ? @Before is executed before each test, so what is the difference with calling init() inside each method ? It is just cleaner I think (meaning that I do a setup and verify it is done correctly). And I do not want to use assume otherwise the test will be ignored.
what would be otherwise a best way to do that in my case ?